python/cudf/cudf/core/series.py

```diff
@@ -1390,7 +1374,7 @@ def sort_index(self, ascending=True):
         """Sort by the index.
         """
         inds = self.index.argsort(ascending=ascending)
-        return self.take(inds)
+        return self.iloc[inds]
```

Comment:
Why did we have to change this? Take is also positional as opposed to based on index values.

Reply:
May be I have to update it [here](https://github.com/rapidsai/cudf/blob/a45da4769e475fa7bbb5168c1cf2bf9f5df0c72f/python/cudf/cudf/core/series.py#L485)

Reply:
Yup, looks like that needs to be updated and we should probably add a test to ensure we're properly gathering positionally for take as well.